presto-main/src/test/java/io/prestosql/operator/scalar/date/TestExtract.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar.date;
+
+import io.prestosql.Session;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.sql.query.QueryAssertions;
+import io.prestosql.testing.TestingSession;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestExtract
```

Comment:
Please see https://github.com/prestosql/presto/pull/5652

Reply:
I know. That's what prompted me to look into this after realizing like `EXTRACT(HOUR FROM DATE '...')` don't fail as they should.